Static analyzer warning fixes

rsanchezsaez opened the pull request:
This gets rid of the static analyzer warning fixes mentioned on [Issue #150](https://github.com/pixelglow/ZipZap/issues/150).

pixelglow commented:
Thanks for stepping up to look at this!

For the `ZZArchive` warnings, prefer using cast to `id _Nonnull` as detailed here:
http://clang-analyzer.llvm.org/faq.html#nullability_intentional_violation

`NSAssert(NO...` is not as obvious in meaning. You can keep the assert comment string as a regular `//` comment.

For the `ZZInflateInputStream` warning, returning `nil` is actually valid for e.g. corrupted compressed data. The `NSAssert` would actually trigger for some user data. So the right thing to do is to annotate the return `(NSData*)` as `(nullable NSData*)`.

Fixing the explicit initialisation of ivars looks good.

Please make the changes above and rebase those changes to squash them with the commit `[Misc] Add asserts to avoid static analyzer warnings`. I'll merge it down then.

rsanchezsaez committed:
[ZZArchive] No need to explicitly initialize instance variables to nil

This avoids a static analyzer warning. See here for more info: http://stackoverflow.com/a/990826/269753

rsanchezsaez committed:
[ZZArchiveEntry] Make intentional nullability violations explicit

Avoids static analyzer warnings. This code will never run because ZZArchiveEntry is an abstract class. More details here: http://clang-analyzer.llvm.org/faq.html#nullability_intentional_violation

rsanchezsaez committed:
[ZZInflateInputStream] Make -decompressData:withUncompressedSize: return type nullable

Avoids a static analyzer warning.

rsanchezsaez commented:
@pixelglow Done!

pixelglow commented:
Excellent! Great stuff.
